src/services/BalanceUtilities.tsx

```diff
@@ -7,21 +7,22 @@ import KiltToken from '../components/KiltToken/KiltToken'
 import * as Balances from '../state/ducks/Balances'
 import * as Wallet from '../state/ducks/Wallet'
 import PersistentStore from '../state/PersistentStore'
+import errorService from './ErrorService'
 import { IContact, IMyIdentity } from '../types/Contact'
-import { notify, notifySuccess } from './FeedbackService'
+import { notify, notifySuccess, notifyFailure } from './FeedbackService'
 
-const KILT_COIN = 1
-const KILT_MICRO_COIN = 1_000_000
-const KILT_FEMTO_COIN = '1000000000000000'
+const KILT_COIN = new BN(1)
+const KILT_MICRO_COIN = new BN(1_000_000)
+const KILT_FEMTO_COIN = new BN('1000000000000000')
 
 // cost of a chain transaction
-const TRANSACTION_FEE = 1 * KILT_COIN
+const TRANSACTION_FEE = new BN(1).mul(KILT_COIN)
 
 // any balance below this will we purged
-const MIN_BALANCE = 1 * KILT_COIN
+const MIN_BALANCE = new BN(1).mul(KILT_COIN)
 
 // initial endowment for automatically created accounts
-const ENDOWMENT = 30 * KILT_COIN
+const ENDOWMENT = new BN(30).mul(KILT_COIN)
```

Comment:
Are you sure, that these all have to be BN, now?

Reply:
These don't need to be a big number, but it is easier to handle the Maths because at some point it becomes a BN.

Reply:
I have updated it and cleaned it up.